### PR TITLE
build: Reorg example kustomizations

### PR DESCRIPTION
**Pull request by jimmidyson**
Reorganised the example kustomization following @deepakm-ntnx's comments.

### CODE REVIEW

**Review by dkoshkin — APPROVED** [review]
Thanks, good changes

File: hack/examples/sync.sh [hunk 1]
```diff
@@ -19,93 +19,26 @@ mkdir -p "${EXAMPLE_CLUSTERCLASSES_DIR}"
...
-                                        and .kind == "Cluster"
-                                        and .spec.topology.variables[0].value.addons.cni.provider == "Cilium"
-                                        and .spec.topology.variables[0].value.addons.cni.strategy == "HelmAddon"
-                                      )' >"${EXAMPLE_CLUSTERS_DIR}/aws-cluster-cilium-helm-addon.yaml"
-    ) \
-    >/dev/null
-
-#shellcheck disable=SC2016
-sed -i'' 's/^  name: .\+$/  name: ${CLUSTER_NAME}/' "${EXAMPLE_CLUSTERS_DIR}"/*.yaml
+kustomize build ./hack/examples/overlays/clusterclasses/aws >"${EXAMPLE_CLUSTERCLASSES_DIR}/aws-cluster-class.yaml"
```
Comment (deepakm-ntnx) [inline hack/examples/sync.sh:22]: line 22-23 and 26 are doing something similar?

Comment (jimmidyson) [reply hack/examples/sync.sh:22]: Oops yeah I forgot to remove these lines when I moved them into the for loop. Will fix up tomorrow. Thanks!

Comment (jimmidyson) [reply hack/examples/sync.sh:22]: Done! Thanks again for pointing this out.

**Pull request merged**